### PR TITLE
Merging 867

### LINKED ISSUE #866: CLI: `config` returns incorrect error
### What version are you using?
soroban 0.9.4 (76cacc7cf885bd7d37756735fab332351d893406)
soroban-env 0.0.17 (400d806387140553e4e685d232deb3a807ec0e36)

### What did you do?

If you don't provide command `address` to `soroban config identity`, then incorrect error appears.

### What did you expect to see?
"Command not recognized; available commands..."

### What did you see instead?
```bash
$ soroban config identity address alice
GCH...
$ soroban config identity  alice        # notice missing `address`
error: no such command: `config`

        View all installed plugins with `soroban --list`
```
There must be a bug in a way the command chain is parsed.

### PR DESCRIPTION
**Pull request by paulbellamy**
Updating and merging #867 

Closes #867
Fixes #866

### CODE REVIEW

File: cmd/crates/soroban-test/tests/it/plugin.rs [hunk 1]
```diff
@@ -4,32 +4,40 @@ is correct. The PATH environment variable is set to include the target/bin
...
         .stdout(predicates::str::contains("hello"));
 }
+
+fn get_paths() -> OsString {
+    // Get the current working directory
+    let current_dir = std::env::current_dir().unwrap();
+    // Create a path to the target/bin directory
+    let target_bin_path = current_dir.join("../../../target/bin");
+    // Get the current PATH environment variable
+    let path_key = std::env::var_os("PATH").unwrap();
```
Comment (tsachiherman) [inline cmd/crates/soroban-test/tests/it/plugin.rs:39]: could you please test this with an environment that lacks PATH ? The method would return none, and I'm not sure if split_paths would take that..

**Review by tsachiherman — COMMENTED** [review]
see comments.

File: cmd/soroban-cli/src/commands/plugin.rs [hunk 2]
```diff
@@ -19,18 +21,31 @@ pub enum Error {
...
 pub fn run() -> Result<(), Error> {
     let (name, args) = {
         let mut args = std::env::args().skip(1);
         let name = args.next().ok_or(Error::MissingSubcommand)?;
         (name, args)
     };
+
+    if Root::command().get_subcommands().any(|c| {
+        let sc_name = c.get_name();
+        sc_name.starts_with(&name) || strsim::jaro(sc_name, &name) >= SUBCOMMAND_TOLERANCE
```
[the remark this answers is not in the document]
Comment (willemneal) [reply cmd/soroban-cli/src/commands/plugin.rs:38]: Sure!

**Comment by tsachiherman** [comment]
@paulbellamy I believe that @willemneal made some updates to his branch based on my feedback above. could you merge that here ?

**Comment by paulbellamy** [comment]
Merged willem's branch instead

**Pull request closed without merging**